Mta browse page mvp functionality

MarkAlperin opened the pull request.

MarkAlperin committed:
merge from ian
Merge branch 'ian' of https://github.com/GreeneOcean/Project_Greene into ian

MarkAlperin committed:
two map options functional, choosing between them

MarkAlperin committed:
Pins are rendered, working on Items/List and ItemCard

MarkAlperin committed:
list and map functional and linked

MarkAlperin committed:
implemented filter functionality

MarkAlperin committed:
added button to selected item for page navigation, also fixed bug with distance filter and removed some comments/logs/extraneous code

MarkAlperin committed:
add package.json changes

MarkAlperin committed:
big merge of browsePage
Merge branch 'dev' of https://github.com/GreeneOcean/Project_Greene into mta_browse_page_mvp_functionality

MarkAlperin committed:
final touches for merge

MarkAlperin committed:
deleted bundle.LISCENESE

7socks commented on client/src/browse/BrowsePage.js:
```diff
@@ -0,0 +1,68 @@
+
+const Container = styled.div`
+  display: flex;
+  justify-content: space-evenly;
```
Reminder to move this file's functionality into the existing Browse.js file in src/pages for your next PR

MarkAlperin replied:
Will do in my "integration" PR

7socks commented on client/src/browse/filterBar/FilterBar.js:
```diff
@@ -0,0 +1,84 @@
+  const filterItems = () => {
+    let filteredItems = props.itemData;
+    if (document.getElementById("distance").value !== "--Select Distance--") {
+      filteredItems = filteredItems.filter((item) => {
```
Is there a reason you're looking up elements on the DOM instead of tracking with state?

MarkAlperin replied:
was just making it run, can change later. call it tech-debt for now?

7socks commented on client/src/browse/maps/Map.js:
```diff
@@ -0,0 +1,57 @@
+
+import MapMarker from "./MapMarker.js";
+require("dotenv").config();
+
```
Does this work for you? I had to not use dotenv in my code.

Doobss replied:
@7socks I used env-cmd yesterday, instead of importing a file it just runs a file in the npm script which in turn allows all downstream processes access to those env's defined in the file. I think the npm script "build:staging" uses the env-cmd if you want to look.

MarkAlperin replied:
Removed dotenv, will replace with env-cmd in "integration" PR

7socks commented on client/src/browse/maps/MapMarker.js:
```diff
@@ -0,0 +1,36 @@
+import MarkerCard from "./MarkerCard.js";
+import { RiMapPin2Fill } from "react-icons/ri";
+// import {FaMapMarkerAlt } from "react-icons/fa";
+const MapMarker = (props) => {
```
Remove commented out lines unless you will need them later

MarkAlperin replied:
Comment removed

7socks commented:
Approving to keep things moving, but do look at the comments

MarkAlperin committed:
made ProductList and ProductCard for HomeItemsWidget

MarkAlperin committed:
completed dummyData dummyFunctionality  widget (no state, no item click functionality)

MarkAlperin committed:
text display acceptable with dummyData

Doobss commented on client/src/browse/BrowsePage.js:
```diff
@@ -0,0 +1,68 @@
+        setFilteredItems(temp);
+      });
+  }, []);
+
```
@MarkAlperin I have some api fetch functions built out in the api folder. We should get together and talk about fetching your data. 

Something like this
```
function get(endpoint, queries) {
  return runFetch(...buildGetOptions(endpoint, queries))
}

function getData(path, queries) {
  return get('/data' + path, queries)
}

function getDonations(queries) {
  return getData('/local/donations', queries)
}
```

I also have the same for getting user location.

MarkAlperin replied:
Removed old fetch request, will run newer query or get data from Context in "integration" PR

Doobss commented on client/src/browse/BrowsePage.js:
```diff
@@ -0,0 +1,68 @@
+        setFilteredItems={setFilteredItems}
+      />
+      <Container>
+        <Map
```
@MarkAlperin just from personal experience a bit more specific names for styled components can help in readability later on. Especially if they are offloaded to a different file. So I would personally name 
```
<Container>
to 
<MapContainer>
```
just to make it a bit more explicit. Not a request or requirement, just sharing my experience.

MarkAlperin replied:
replaced with a more descriptive name for styled-component

Doobss commented on client/src/browse/filterBar/FilterBar.js:
```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+
+const FilterBar = (props) => {
```
@MarkAlperin I would recommend destructuring props to reduce property lookups and enhance readability (imo).
```
const FilterBar = (props) => {

becomes

const FilterBar = ({ itemData, setFilteredItems,  setSelectedItem }) =>  { 

(I think I got all of the props)
```
and then remove "props." below where ever it is accessed.

MarkAlperin replied:
can I consider this tech-debt for now and return when I have time/non-broken components?

Doobss commented on client/src/browse/list/ItemCard.js:
```diff
@@ -0,0 +1,49 @@
+import { BsCardImage } from "react-icons/bs";
+
+const ItemCard = (props) => {
+  const clickHandler = () => {
```
@MarkAlperin just commenting about destructuring again. Would recommend the same here.

Doobss commented on client/src/browse/list/ItemCard.js:
```diff
@@ -0,0 +1,49 @@
+      onClick={clickHandler}
+      style={{
+        backgroundColor: props.selectedItem === props.item.id ? "red" : "blue",
+      }}
```
@MarkAlperin I see that you're using inline styling to conditionally render, Im not sure if you're aware that styled comp's have this functionality built in.


```
//Im going to assume props has been destructered 
//so selectedItem === props.selectedItem  &&  item === props.item

<Container
  id={item.id}
  onClick={clickHandler}
  isSelected={selectedItem === item.id}
>

const Container = styled.div`
  min-width: 150px;
  max-width: 150px;
  height: 150px;
  margin: 10px;
  cursor: pointer;
  background-color: ${({ isSelected } => isSelected ? "red" : "blue" )};
`;  
```

This might be an approach I would take, but Im sure both work. I would like to hear your thoughts. Whether it is on here or later today.

MarkAlperin replied:
changing now

Doobss commented on client/src/browse/list/ItemList.js:
```diff
@@ -0,0 +1,42 @@
+  display: flex;
+  overflow: scroll;
+  display: flex;
+  flex-wrap: wrap;
```
@MarkAlperin with the double flex lol.

MarkAlperin replied:
removed

7socks replied:
weird flex but ok

Doobss commented on client/src/browse/list/ItemList.js:
```diff
@@ -0,0 +1,42 @@
+import ItemCard from "./ItemCard.js";
+
+const ItemList = (props) => {
+
```
@MarkAlperin Im sure you know what Im going to say already lol

```
const ItemList = ({ items, selectedItem, setSelectedItem }) => {
```
As a thanks I believe that should get all the props

edit: there are more below, Ill spare the extra comments and assume the same for the rest, thanks.

Doobss reviewed (changes requested):
Good stuff Mark. Just would request that you'd destructor props for the components. If you'd  like to discuss anything let me know.

MarkAlperin committed:
pulling from mta_browse_page_mvp_functionality to make requested changes
Merge branch 'mta_browse_page_mvp_functionality' of https://github.com/GreeneOcean/Project_Greene into mta_browse_page_mvp_functionality

MarkAlperin committed:
make BrowsePage container name more descriptive

MarkAlperin committed:
removed process.env import

MarkAlperin committed:
changed google map api key from process.env to a temp string placeholder

MarkAlperin committed:
removed comment from MapMarker.js

MarkAlperin committed:
removed old fetch request, need to replace later

MarkAlperin committed:
replace styled-component names with more descriptive names

MarkAlperin commented:
I made the minor changes,  I would prefer to make the destructuring changes  and filtering w/o DOM elements changes during the "integration" PR so that I am not introducing new bugs. I don't have access to data atm and will be coding blind. Happy to discuss ofc

MarkAlperin committed:
change conditional styles functionality to use styled-components

MarkAlperin committed:
removed redundant css line

MarkAlperin committed:
destructured props for these files

MarkAlperin committed:
destructure props

MarkAlperin commented:
I had a little extra time this morning  and I got those props destructures done :)

Doobss reviewed (dismissed):
Thanks MArk

MarkAlperin committed:
removed code from pages/Item.js

MarkAlperin committed:
Merge branch 'dev' into mta_browse_page_mvp_functionality

MarkAlperin committed:
fix merge conflict

MarkAlperin committed:
connecting components to global state

MarkAlperin committed:
moved browse directory into the components directory

MarkAlperin committed:
implement View button functionality on item cards

MarkAlperin committed:
merge from changes made on github
Merge branch 'mta_browse_page_mvp_functionality' of https://github.com/GreeneOcean/Project_Greene into mta_browse_page_mvp_functionality

MarkAlperin committed:
delete item/ from this branch
